tools/shared.py

```diff
@@ -1167,6 +1165,9 @@ def load(self, args=[]):
           declare = re.sub(r'([\w\d]+)\s*=\s*(.+)', r'self.attrs["\1"]=\2;', args[i+1])
           exec declare
 
+      if get_llvm_target() == WASM_TARGET:
```

Comment:
IIRC, This code is running when the Settings object is loaded.  Its not possible to access the Settings before this time I think (you will get attribute error if somehow manage to do that, just like do if you try to access a non-existent setting).